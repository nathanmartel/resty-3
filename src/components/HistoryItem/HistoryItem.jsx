import React from 'react';
import PropTypes from 'prop-types';

const HistoryItem = ({ method, url, index, onLoadHistoryItemClick }) => {
  return (
    <>
      <p>{method} {url}</p>
      <button onClick={() => onLoadHistoryItemClick(index)}>Load</button>
      <hr />
    </>
  );
};

HistoryItem.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  onLoadHistoryItemClick: PropTypes.func.isRequired
};

export default HistoryItem;
