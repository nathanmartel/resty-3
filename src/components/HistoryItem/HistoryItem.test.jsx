import React from 'react';
import { shallow } from 'enzyme';
import HistoryItem from './HistoryItem';

describe('HistoryItem component', () => {
  it('renders HistoryItem', () => {
    const wrapper = shallow(<HistoryItem 
      method={'GET'}
      url={'http://www.test.com'}
      index={1}
      onLoadHistoryItemClick={() => {}} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
