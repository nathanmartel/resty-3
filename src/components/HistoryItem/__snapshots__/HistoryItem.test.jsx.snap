// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component renders HistoryItem 1`] = `
<Fragment>
  <p>
    GET
     
    http://www.test.com
  </p>
  <button
    onClick={[Function]}
  >
    Load
  </button>
  <hr />
</Fragment>
`;
